go/types/list.go

```diff
@@ -175,7 +206,10 @@ func (l List) Iter(f listIterFunc) {
 
 type listIterAllFunc func(v Value, index uint64)
 
+// IterAll iterates over the list and calls f for every element in the list. Unlike Iter there is no
+// way to stop the iteration and all elements are visited no matter what.
 func (l List) IterAll(f listIterAllFunc) {
+	// TODO: Consider removing this and have Iter behave like IterAll.
```

Comment:
I agree, and have thought the same thing. IterAll saves you an extra line (a return statement).

Though I think an issue, rather than a TODO, is less likely to get lost.